auth-api/src/auth_api/services/affiliation.py

```diff
@@ -128,18 +131,65 @@ def create_affiliation(org_id, business_identifier, pass_code=None, token_info:
         if affiliation is not None:
             raise BusinessException(Error.DATA_ALREADY_EXISTS, None)
 
-        # Retrieve entity name from Legal-API and update the entity with current name
-        # TODO: Create subscription to listen for future name updates
-        current_app.logger.debug('<create_affiliation sync_name')
-        entity.sync_name(bearer_token)
-
         affiliation = AffiliationModel(org_id=org_id, entity_id=entity_id)
         affiliation.save()
+
         entity.set_pass_code_claimed(True)
-        current_app.logger.debug('<create_affiliation affiliated')
 
         return Affiliation(affiliation)
 
+    @staticmethod
+    def create_new_business_affiliation(org_id, business_identifier=None,  # pylint: disable=too-many-arguments
+                                        email=None, phone=None, token_info: Dict = None, bearer_token: str = None):
+        """Initiate a new incorporation."""
+        # Validate if org_id is valid by calling Org Service.
+        current_app.logger.info(f'<create_affiliation org_id:{org_id} business_identifier:{business_identifier}')
+        affiliation_model = None
+
+        if not email and not phone:
+            raise BusinessException(Error.NR_INVALID_CONTACT, None)
+
+        org = OrgService.find_by_org_id(org_id, token_info=token_info, allowed_roles=CLIENT_AUTH_ROLES)
+        if org is None:
+            raise BusinessException(Error.DATA_NOT_FOUND, None)
+
+        entity = EntityService.find_by_business_identifier(business_identifier, skip_auth=True)
+        # If entity already exists and is already affiliated to an org, throw error
+        if entity and entity.as_dict()['passCodeClaimed']:
+            raise BusinessException(Error.NR_CONSUMED, None)
+
+        # Call the legal-api to verify the NR details
+        nr_json = Affiliation._get_nr_details(business_identifier, bearer_token)
+
+        if nr_json:
+            status = nr_json.get('state')
+            nr_phone = nr_json.get('applicants').get('phoneNumber')
+            nr_email = nr_json.get('applicants').get('emailAddress')
+
+            if status not in ('APPROVED', 'CONDITIONAL'):
+                raise BusinessException(Error.NR_NOT_APPROVED, None)
+
+            if (phone and phone != nr_phone) or (email and email != nr_email):
+                raise BusinessException(Error.NR_INVALID_CONTACT, None)
+
+            # If all good create the business and affiliation
+            if not entity:
+                name = next(
```

Comment:
hm..whats the logic in here?i was trying to understand it..may be a comment will help in here

Reply:
Done